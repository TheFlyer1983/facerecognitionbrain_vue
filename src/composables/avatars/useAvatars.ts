import { Ref, ref } from 'vue';

function useAvatars() {
  const avatar: Ref<string> = ref('');

  async function createAvatar(name: string) {
    const initials = name.charAt(0);

    avatar.value = initials;
  }

  return { avatar, createAvatar };
}

export default useAvatars;
