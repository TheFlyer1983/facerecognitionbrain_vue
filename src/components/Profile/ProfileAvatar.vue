<script setup lang="ts">
import useAvatars from '@/composables/avatars/useAvatars';
import { onMounted } from 'vue';

const props = defineProps<{
  name: string;
}>();

const { avatar, createAvatar } = useAvatars();

onMounted(() => {
  createAvatar(props.name);
});
</script>

<template>
  <div class="avatar d-flex justify-content-center align-items-center">
    {{ avatar }}
  </div>
</template>

<style lang="scss" scoped>
.avatar {
  border-radius: 100%;
  border: 1px solid black;
  display: inline-block;
  height: 4rem;
  width: 4rem;
  font-size: 2.25rem;
  font-weight: bold;
  background-color: #33b2f6;
}
</style>
