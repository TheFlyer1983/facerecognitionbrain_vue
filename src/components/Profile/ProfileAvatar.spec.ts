import { ref } from 'vue';
import { shallowMount, VueWrapper } from '@vue/test-utils';
import ProfileAvatar from './ProfileAvatar.vue';
import useAvatars from '@/composables/avatars/useAvatars';

vi.mock('@/composables/avatars/useAvatars');

const mockUseAvatars = vi.mocked(useAvatars);
const createAvatarMock = vi.fn();

const mockProps = {
  name: 'Paul'
};

describe('Given the `ProfileAvatar` component', () => {
  const render = () =>
    shallowMount(ProfileAvatar, {
      props: {
        name: mockProps.name
      }
    });
  let wrapper: VueWrapper<InstanceType<typeof ProfileAvatar>>;

  beforeAll(() => {
    mockUseAvatars.mockReturnValue({
      avatar: ref('P'),
      createAvatar: createAvatarMock
    });
  });

  describe('when it is rendered', () => {
    beforeEach(() => {
      wrapper = render();
    });

    it('should match the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should call the `createAvatar` function', () => {
      expect(createAvatarMock).toHaveBeenCalledWith(mockProps.name);
    });
  });
});
