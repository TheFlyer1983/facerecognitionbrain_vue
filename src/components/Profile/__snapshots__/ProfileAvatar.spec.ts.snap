// Vitest Snapshot v1

exports[`Given the \`ProfileAvatar\` component > when it is rendered > should match the snapshot 1`] = `
<div
  class="avatar d-flex justify-content-center align-items-center"
  data-v-cd0cfdf7=""
>
  P
</div>
`;
